### PR TITLE
Fix 463 introduce yamllint to improve code consistency

### PR DESCRIPTION
**Pull request by javydekoning**
*Issue #, if available:*
#463 

*Description of changes:*
- Implemented yamllint configuration file.
- Fixed all yamllint warnings and errors.

To run do: `yamllint .`

**NOTE:** this has not been added to CI yet. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by StewartW — APPROVED** [review]
Nice one!! 🙌

**Review by StewartW — DISMISSED** [review]

File: .yamllint [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  key-duplicates: enable
+  key-ordering: disable
+  line-length: disable
+  new-line-at-end-of-file: disable
+  new-lines: enable
+  octal-values: disable
+  quoted-strings: disable
+  trailing-spaces: enable
+  truthy:
+    level: warning
```
Comment (sbkok) [inline .yamllint:31]: These can be annoying mistakes too, I propose to report these as errors too:
```suggestion
    level: error
```

File: .yamllint [hunk 2]
```diff
@@ -0,0 +1,31 @@
+---
+yaml-files:
+  - '*.yaml'
+  - '*.yml'
+  - '.yamllint'
+
+rules:
+  braces: enable
```
Comment (sbkok) [inline .yamllint:8]: Lets prefer the use of braces only for empty objects.
```suggestion
  braces:
    forbid: non-empty
    min-spaces-inside-empty: 0
    max-spaces-inside-empty: 0
```

File: .yamllint [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  colons: enable
+  commas: enable
+  comments:
+    level: warning
+  comments-indentation:
+    level: warning
+  document-end: disable
+  document-start: disable
+  empty-lines: enable
+  empty-values: disable
```
Comment (sbkok) [inline .yamllint:19]: Adding `float-values` rule too:
```suggestion
  empty-values: disable
  float-values:
    forbid-inf: true
    forbid-nan: true
    forbid-scientific-notation: true
    require-numeral-before-decimal: true
```

File: requirements.txt [hunk 4]
```diff
@@ -8,3 +8,4 @@ pytest~=6.2.5
 pyyaml>=5.4.1
 schema~=0.7.5
 tox==3.24.4
+yamllint~=1.26.3
```
Comment (sbkok) [inline requirements.txt:11]:
```suggestion
yamllint~=1.27.1
```

**Review by sbkok — CHANGES_REQUESTED** [review]
Great improvement, thanks for making these consistent!

File: .yamllint (hunk 1 above)
Comment (javydekoning) [reply .yamllint]: Thanks, added.

**Review by sbkok — APPROVED** [review]
LGTM, thanks!

**Review by StewartW — APPROVED** [review]

**Pull request merged**